notes/advanced-data-types.md

```diff
@@ -0,0 +1,530 @@
+# Advanced Data Types
+
+## Introduction
+
+Here we will discuss types that are unique to C++. This includes `stucts`, `enums` and `unions`. We will also discuss `namespace`, which is not a type but rather an entity that holds variables and functions.
```

Comment:
structs and enums are also found in C/Java so I wouldn't necessarily say unique

also small typo `struct`